File: chess-core/src/main/kotlin/net/zomis/chess/core/ChessTile.java
package net.zomis.chess.core;

public class ChessTile {

    private final ChessPosition position;
    private Player player;
    private Piece piece;

    public ChessTile(int x, int y) {
        this.position = new ChessPosition(x, y);
    }

    @Override
    public String toString() {
        String pl = player == null ? "_" : player.toString();
        String piece = this.piece == null ? "_" : this.piece.toString();
        return pl + piece;
    }

    public void set(Player owner, Piece piece) {
        this.player = owner;
        this.piece = piece;
    }

    public Player getOwner() {
        return player;
    }

    public Piece getPiece() {
        return piece;
    }

    public ChessPosition getPosition() {
        return position;
    }

    public void clear() {
        this.player = null;
        this.piece = null;
    }

    public void set(ChessTile tile) {
        this.set(tile.getOwner(), tile.getPiece());
    }

    public String getFullName() {
        if (player == null || piece == null) {
            return "EMPTY";
        }
        return player.name() + "_" + piece.name();
    }
}

File: chess-core/src/main/kotlin/net/zomis/chess/core/ChessTile.kt
package net.zomis.chess.core

class ChessTile(x: Int, y: Int) {

    val position: ChessPosition
    var owner: Player? = null
        private set
    var piece: Piece? = null
        private set

    val fullName: String
        get() = if (owner == null || piece == null) {
            "EMPTY"
        } else owner!!.name + "_" + piece!!.name

    init {
        this.position = ChessPosition(x, y)
    }

    override fun toString(): String {
        val pl = if (owner == null) "_" else owner!!.toString()
        val piece = if (this.piece == null) "_" else this.piece!!.toString()
        return pl + piece
    }

    operator fun set(owner: Player?, piece: Piece?) {
        this.owner = owner
        this.piece = piece
    }

    fun clear() {
        this.owner = null
        this.piece = null
    }

    fun set(tile: ChessTile) {
        this[tile.owner] = tile.piece
    }
}
